### ISSUE TITLE
Support for ESP32S3 TTGO T-Display S3

### ISSUE THREAD

**Issue by g2gps** [apache/nuttx#7803]
Hello,

I'm attempting to get Nuttx up and running with LVGL on the ESP32S3 TTGO T-Display S3.  ([product link](https://github.com/Xinyuan-LilyGO/T-Display-S3)). As far as I'm aware there isn't currently a configuration supplied for this board.

So far, I've been able to use `esp32s3-devkit:nsh`, as a base and enabling the USB-serial driver to use for nsh. Going forward, I was hoping to use the `ttgo_t_display_esp32:lvgl_fb` as a base for enabling support for the display, given they use the same `ST7789` display driver.

Perhaps there's a better way to approach this, or somebody has gone through this process previously? I'm quite new to Nuttx, so I may be overlooking something obvious.

**Comment by acassis** [comment]
Hi @g2gps I receive this same board that I bought from aliexpress, so I want to test it too. AFAIK ESP32S3 TTGO T-Display S3 uses LCD in parallel mode instead of SPI mode.

**Comment by acassis** [comment]
Hi @hanfengcan sorry I didn't see your message before. Seems like the LCD parallel support was included into mainline.
Maybe @fdcavalcanti or @eren-terzioglu can confirm it.

**Comment by eren-terzioglu** [comment]
> Hi [@hanfengcan](https://github.com/hanfengcan) sorry I didn't see your message before. Seems like the LCD parallel support was included into mainline. Maybe [@fdcavalcanti](https://github.com/fdcavalcanti) or [@eren-terzioglu](https://github.com/eren-terzioglu) can confirm it.

There are some LCD parallel options  [in Kconfig](https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32s3/Kconfig#L2559) and used in [some implementation](https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32s3/esp32s3_lcd.c) that powers up `esp32s3-lcd-ev:lvgl` defconfig but I don't remember any implementation using I2S. It is possible to implement but not on our plans right now

**Comment by acassis** [comment]
@eren-terzioglu I think I2S was for ESP32, that share that same interface for Audio and Camera, maybe he made some confusion about it

**Comment by hanfengcan** [comment]
just esp32s3 support，esp32 is not

**Comment by eren-terzioglu** [comment]
> just esp32s3 support，esp32 is not

Yes, seems we don't have any paralled lcd support for esp32. ESP controller driver needs to be implemented before than display driver. 

If you would like to implement a solution feel free to ask any question we would like to help but implementing this feature is not in our plans

**Comment by acassis** [comment]
> > Hi [@hanfengcan](https://github.com/hanfengcan) sorry I didn't see your message before. Seems like the LCD parallel support was included into mainline. Maybe [@fdcavalcanti](https://github.com/fdcavalcanti) or [@eren-terzioglu](https://github.com/eren-terzioglu) can confirm it.
> 
> There are some LCD parallel options [in Kconfig](https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32s3/Kconfig#L2559) and used in [some implementation](https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32s3/esp32s3_lcd.c) that powers up `esp32s3-lcd-ev:lvgl` defconfig but I don't remember any implementation using I2S. It is possible to implement but not on our plans right now

@eren-terzioglu looking esp32s3-lcd-ev/src/esp32s3_lcd.c seems like it is using SPI, strange it was implemented as a board driver instead of a generic SPI driver

**Comment by eren-terzioglu** [comment]
> > > Hi [@hanfengcan](https://github.com/hanfengcan) sorry I didn't see your message before. Seems like the LCD parallel support was included into mainline. Maybe [@fdcavalcanti](https://github.com/fdcavalcanti) or [@eren-terzioglu](https://github.com/eren-terzioglu) can confirm it.
> > 
> > 
> > There are some LCD parallel options [in Kconfig](https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32s3/Kconfig#L2559) and used in [some implementation](https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32s3/esp32s3_lcd.c) that powers up `esp32s3-lcd-ev:lvgl` defconfig but I don't remember any implementation using I2S. It is possible to implement but not on our plans right now
> 
> [@eren-terzioglu](https://github.com/eren-terzioglu) looking esp32s3-lcd-ev/src/esp32s3_lcd.c seems like it is using SPI, strange it was implemented as a board driver instead of a generic SPI driver

I checked file history and did not find any big difference. Here are the [commits](https://github.com/apache/nuttx/commits/master/boards/xtensa/esp32s3/esp32s3-lcd-ev/src/esp32s3_lcd.c).
